software/duet_config_files/duet2/rrf3/sys/homex.g

```diff
@@ -1,7 +1,17 @@
 ; Home X Axis
 
```

Comment:
I'd recommend adding a comment here so we don't remove have to ask ourselves why these two conditionals are here later. Something like this would work:
```
; In case homex.g is called in isolation, ensure 
; (1) U axis is homed (which performs tool detection and sets machine tool state to a known state) and 
; (2) No tools are loaded.
; Ask for user-intervention if either case fails.
```
These are great checks to prevent the user from hurting their machine from a cold boot. Now that we can add conditionals, let's additionally add checks to make sure:
(3) y is homed and
(4) y value of machine is out of the parking region, i.e: ```move.axes[2] < 305```, so we don't crash into tools while homing x

Reply:
3. Done
4. Done

Passed Tests:
no tool, safe area, y homed
tool, safe area, y homed
no tool, danger area, y homed
tool, danger area, y homed
no tool, safe area, y not homed
tool, safe area, y not homed
no tool, danger area, y not homed
tool, danger area, y not homed